modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/RoleResourceV2Manager.java

```diff
@@ -328,6 +325,179 @@ public SCIMResponse updateWithPATCHRole(String id, String patchRequest, RoleV2Ma
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceV2Schema();
+            Map<String, Boolean> requestAttributes = ResourceManagerUtil.getAllAttributeURIs(schema);
+
+            List<PatchOperation> opList = getDecoder().decodeRequest(patchRequest);
+
+            if (!isUpdateAllUsersOperationFound(opList)) {
+                return updateWithPatchOperationsMeta(id, opList, roleManager, schema, encoder);
+            }
+
+            RoleV2 oldRole = getOldRoleforV3Api(id, roleManager, requestAttributes);
+            // Make a copy of original role. This will be used to restore to the original condition if failure occurs.
+            RoleV2 originalRole = (RoleV2) CopyUtil.deepCopy(oldRole);
+            RoleV2 patchedRole = doPatchRole(oldRole, schema, patchRequest);
+            RoleV2 updatedRole = roleManager.updateRoleMeta(originalRole, patchedRole);
+            return getScimResponse(encoder, updatedRole);
+        } catch (NotFoundException | BadRequestException | NotImplementedException | CharonException | ConflictException
+                 | InternalErrorException e) {
+            return AbstractResourceManager.encodeSCIMException(e);
+        } catch (RuntimeException e) {
+            CharonException ex = new CharonException("Error in performing the patch operation on role resource.", e);
+            return AbstractResourceManager.encodeSCIMException(ex);
+        }
+    }
+
+    @Override
+    public SCIMResponse updateUsersWithPUTRole(String id, String putRequest, RoleV2Manager roleManager) {
+
+        try {
+            validateManager(roleManager);
+            JSONEncoder encoder = getEncoder();
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceV3UserSchema();
+            Map<String, Boolean> requestAttributes = ResourceManagerUtil.getAllAttributeURIs(schema);
+            RoleV2 role = decodeRole(putRequest, schema);
+            RoleV2 updatedRole;
+
+            // Retrieve the old object.
+            RoleV2 oldRole = getOldRoleforV3Api(id, roleManager, requestAttributes);
+            RoleV2 newRole = (RoleV2) ServerSideValidator.validateUpdatedSCIMObject(oldRole, role, schema);
+            updatedRole = roleManager.updateUsersRole(oldRole, newRole);
+            return getScimResponse(encoder, updatedRole);
+        } catch (NotFoundException | BadRequestException | CharonException | ConflictException | InternalErrorException
+                 | NotImplementedException e) {
+            return encodeSCIMException(e);
+        }
+    }
+
+    @Override
+    public SCIMResponse updateUsersWithPATCHRole(String id, String patchRequest, RoleV2Manager roleManager) {
+
+        try {
+            validateManager(roleManager);
+            JSONEncoder encoder = getEncoder();
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceV3UserSchema();
```

Comment:
SCIM Spec defines some standard schemas, we should not create schemas for all just for the sake of validating the payload

Reply:
Fixed with : https://github.com/wso2/charon/pull/435/commits/adb417717e375a2b27daa8200ee3b635f49e8332